Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -1030,11 +1036,42 @@ HRESULT winhttp_http_task::connect_and_send_async()
 
 NAMESPACE_XBOX_HTTP_CLIENT_END
 
+#if HC_PLATFORM == HC_PLATFORM_GSDK
+typedef DWORD(WINAPI *GetNetworkConnectivityHintProc)(NL_NETWORK_CONNECTIVITY_HINT*);
+#endif
+
 HRESULT Internal_InitializeHttpPlatform(HCInitArgs* args, PerformEnv& performEnv) noexcept
 {
     assert(args == nullptr);
     UNREFERENCED_PARAMETER(args);
 
+#if HC_PLATFORM == HC_PLATFORM_GSDK
+    if (XGameRuntimeIsFeatureAvailable(XGameRuntimeFeature::XNetworking))
+    {
+        HMODULE hModule = LoadLibrary(TEXT("iphlpapi.dll"));
+
+        if (hModule != nullptr)
+        {
+            GetNetworkConnectivityHintProc getNetworkConnectivityHint =
+                (GetNetworkConnectivityHintProc)GetProcAddress(hModule, "GetNetworkConnectivityHint");
+           
+            if (getNetworkConnectivityHint != nullptr)
+            {
+                NL_NETWORK_CONNECTIVITY_HINT connectivityHint;
+                HRESULT hr = HRESULT_FROM_WIN32(getNetworkConnectivityHint(&connectivityHint));
+                bool networkNotReady = connectivityHint.ConnectivityLevel == NetworkConnectivityLevelHintUnknown;
+
+                FreeLibrary(hModule);
+
```

Comment:
FreeLibrary needs to go outside the inner if check and inside   if (hModule != nullptr)